### ISSUE TITLE
Unable to create Analytics identifyUser with EMAIL or SMS or Push notification channel

### ISSUE THREAD

**Issue by pascalito007** [aws-amplify/amplify-js#12721]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

Analytics

### Amplify Categories

analytics

### Environment information

<details>

```
# Put output below this line


```

</details>

### Describe the bug

When using identifyUser from Analytics, I have no options for channel and optout. This is available in the In-App aws-amplify/in-app-messaging but not in @aws-amplify/analytics.
Before migrating to V6 it was working in V5.

### Expected behavior

I should be able to specify channelType and optout in order to identify an analytics pinpoint user.

### Reproduction steps

1. import identifyUser from @aws-amplify/analytics
2. create an EMAIL segment in pinpoint console
3. create a journey that send email to users created using identifyUser based on an event
4. The email is never sent and the user never enter the journey

### Code Snippet

```javascript
import { record, identifyUser } from "@aws-amplify/analytics";

await identifyUser({
          userId: user.userId,
          userProfile: {
            email: "xxxxxxxxx@gmail.com",
          },
        });

        record({ name: "TestEvent" });

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by cwomack** [comment]
@pascalito007, confirmed this is a feature we don't have in v6 out of the box.  I'll label this as a feature request and will review it with our team internally! Thank you for creating this issue and just want to confirm the context here.  It sounds like you're basically wanting to send an email when an event is recorded, right?

**Comment by pascalito007** [comment]
Yes correct @cwomack  I have a pinpoint journey which send EMAIL to EMAIL  channel type segment and SMS to sms channel type segment. This is triggered previously in v5 when using updateEndpoint+Analytics event record. I want the same behavior with identifyUser from Analytics + Analytics Event record.

**Comment by soplan** [comment]
Yeah this is stupid that it's not in v6.. I started a new project but am now considering switching back to v5 because this is a very important missing piece of the puzzle. Can we have an estimate?

**Comment by mohammedsahl** [comment]
+1 We use email as our channel. It's crucial for us. Can't upgrade to v6 without it :/

**Comment by monholm** [comment]
This is also an issue when it comes to push notifications.

We use Pinpoint for push notifications, but don't use `aws-amplify/push-notifications` to handle said notifications on the (react-native) client side (for various reasons outside the scope of this issue).

Not exposing options such as `optOut`, `address` and `channelType` in `identifyUser` from `aws-amplify/analytics` forces us to install `@aws-amplify/rtn-push-notification`.

If all `@aws-amplify/rtn-push-notification` did was to provide us a version of `identifyUser` with the correct interface and maybe some other js functions that we could just ignore, we probably wouldn't care. But `@aws-amplify/rtn-push-notification` comes with a range of native functionality, that breaks our existing setup.

This includes registering an android service with an intent-filter for the `com.google.firebase.MESSAGING_EVENT` action and depending on old kotlin-specific firebase dependencies. Long story short, we can't just install `@aws-amplify/rtn-push-notification` and call it a day.

@cwomack since this was labelled as a feature-request, would you accept a PR that aligns the different `identifyUser` functions? We're currently patching amplify-js to include the options in analytics, but would really prefer to "fix" it upstream.

**Comment by renso3x** [comment]
Facing similar issue with v6

1. Using `IdentifyUser` to update the endpoint:

```javascript
await identifyUser({
  userId: email,
  userProfile: {
    name: (user?.given_name ?? userProfile?.givenName) as string,
    email,
    customProperties: {
      organizationName: [staffSession.oName],
      product: [PRODUCT],
    },
  },
  options: {
    userAttributes: {
      role: [userProfile?.role as string],
    },
  },
});

record({
  name: `${PRODUCT}_USER_SIGNIN`,
});
```
2. This should trigger an event in the journey where the user activity is _USER_SIGNIN, and an email should be sent. However, none of the journey events were triggered. 


I was only able to trigger the event by using the following command:

`aws pinpoint put-events --application-id 7149cbb8XXXXXXXX --events-request file://PutEvents.json`

**Comment by cwomack** [comment]
@renso3x, thank you for the additional context here.  

@monholm, let me review this again with our team internally and get back as soon as possible. Appreciate the patience from everyone following this issue.

**Comment by HemangiVekaria** [comment]
I migrated from this
```
"@aws-amplify/pushnotification": "^3.2.5",
"aws-amplify": "^3.3.1",  
```
to this
```
"@aws-amplify/rtn-push-notification": "^1.2.31",
"aws-amplify": "^6.8.2",
```

Now old version has method to update user token in pinpoint using 
import PushNotification from '@aws-amplify/pushnotification';
PushNotification.updateEndpoint(token);

This will update token to pinpoint for notification. Now i am struggling to update token on pinpoint with latest version of aws-amplify.

Can anyone has idea for this? i am struggling since 4 days. I am migrating to latest version and in old notification work properly but after updating aws-amplify to latest push notification is not working. 

rtn-push-notification with this i am able to get token. Also if from Aws console i add that token in Test Messaging then i am getting notification in device. 

 Please help for this.

**Comment by HuiSF** [comment]
Hi @HemangiVekaria looking at the code:

```ts
PushNotification.updateEndpoint(token);
```

By the token you meant the device token for registering Push Notification correct? With Amplify V6 you should not need a manual step for this. The library handles token registration automatically. You can review the Push Notification [documentation](https://docs.amplify.aws/gen1/react-native/build-a-backend/push-notifications/) for necessary steps.

**Comment by HemangiVekaria** [comment]
> Hi @HemangiVekaria looking at the code:
> 
> ```ts
> PushNotification.updateEndpoint(token);
> ```
> 
> By the token you meant the device token for registering Push Notification correct? With Amplify V6 you should not need a manual step for this. The library handles token registration automatically. You can review the Push Notification [documentation](https://docs.amplify.aws/gen1/react-native/build-a-backend/push-notifications/) for necessary steps.

By default, it sends the token as 'undefined.' I tried the code below, and it successfully sends the token to Pinpoint. Now, I am receiving notifications. I don't know why the token is sent as 'undefined' by default.

```
import {identifyUser} from 'aws-amplify/push-notifications';
identifyUser({
        address: token,
        channelType: IS_IOS ? 'APNS' : 'GCM',
        userId: `${this.props.user.id}`,
        immediate: true,
      });
```
